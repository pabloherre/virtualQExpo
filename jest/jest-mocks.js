import mockAsyncStorage from '@react-native-community/async-storage/jest/async-storage-mock';
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
jest.requireActual('./mocks');
jest.mock('@react-native-community/async-storage', () => mockAsyncStorage);
jest.mock('expo-barcode-scanner');
jest.mock('expo-location');
jest.mock('react-native-maps');
