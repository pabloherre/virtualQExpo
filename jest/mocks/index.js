export * from './appointment.mock';
export * from './appointments.mock';

