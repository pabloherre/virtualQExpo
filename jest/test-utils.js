import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);
function connectedRender(ui, initialState) {
  const store = mockStore(initialState);
  return renderer.create(<Provider store={store}>{ui}</Provider>);
}

function connectedShallow(ui, initialState) {
  const store = mockStore(initialState);
  return shallow(<Provider store={store}>{ui}</Provider>);
}

function connectedMount(ui, initialState, numberOfHoc) {
  const store = mockStore(initialState);
  return mount(<Provider store={store}>{ui}</Provider>);
}

// override connectedRender method
export { connectedRender, connectedShallow, connectedMount };
