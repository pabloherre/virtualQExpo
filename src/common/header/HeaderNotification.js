import React, { Component } from 'react';
import { EvilIcons } from '@expo/vector-icons';

class HeaderNotification extends Component {
  render() {
    return <EvilIcons name="bell" size={36} color="#a1a2a4" />;
  }
}

HeaderNotification.propTypes = {};

export default HeaderNotification;
