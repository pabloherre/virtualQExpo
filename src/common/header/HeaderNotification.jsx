import React, { Component } from 'react';
import { EvilIcons } from '@expo/vector-icons';
import { withTheme } from '../theme/Theme';

class HeaderNotification extends Component {
  render() {
    const {
      theme: { colors }
    } = this.props;
    return <EvilIcons name="bell" size={36} color={colors.text} />;
  }
}

HeaderNotification.propTypes = {};

export default withTheme(HeaderNotification);
