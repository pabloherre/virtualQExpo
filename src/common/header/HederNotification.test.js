import React from 'react';
import renderer from 'react-test-renderer';
import HeaderNotification from './HeaderNotification';
import { connectedRender } from '../../../jest/test-utils';

describe('<HeaderNotification />', () => {
  it('renders correctly', async () => {
    const button = connectedRender(<HeaderNotification />).toJSON();
    expect(button).toMatchSnapshot();
  });
});
