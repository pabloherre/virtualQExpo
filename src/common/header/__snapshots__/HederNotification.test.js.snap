// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderNotification /> renders correctly 1`] = `<Text />`;
